### PR TITLE
chore: Add tooltips and button to Connect Postgresql DB Modal Form

### PR DESCRIPTION
**Pull request by andrewbastian**
### SUMMARY

Add tooltips and edit tooltip dialog on Postgresql connection modal, Add button link on SQLAcademy modal.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screen Shot 2021-06-15 at 1 02 56 PM
Screen Shot 2021-06-15 at 1 01 46 PM
Screen Shot 2021-06-15 at 1 01 40 PM
Screen Shot 2021-06-17 at 10 01 01 PM
Screen Shot 2021-06-17 at 9 58 49 PM

### TESTING INSTRUCTIONS

Visual inspection of changes to modal. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/SqlAlchemyForm.tsx [hunk 1]
```diff
@@ -92,6 +93,30 @@ const SqlAlchemyTab = ({
...
       {t('Test connection')}
     </Button>
+      {/*Button link needs config. Commented code taken from ./index.tsx */}
+      <div css={(theme: SupersetTheme)=> infoTooltip(theme)}>
+          <Button
+              buttonStyle="link"
+              onClick={() =>
+                alert('Fix this link and delete line 101 in ./SqlAlchemyForm.tsx')
+                  
+              //    setDB({
```
Comment (hughhhh) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/SqlAlchemyForm.tsx:103]: Remove these comments

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
  * Licensed to the Apache Software Foundation (ASF) under one
  * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
+placement* distributed with this work for additional information
```
Comment (hughhhh) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:4]: the star should be to the left

File: superset-frontend/src/components/Form/LabeledErrorBoundInput.stories.tsx [hunk 3]
```diff
@@ -20,29 +20,26 @@ import React, { useState } from 'react';
 import LabeledErrorBoundInput, {
   LabeledErrorBoundInputProps,
 } from './LabeledErrorBoundInput';
-
```
Comment (lyndsiWilliams) [inline superset-frontend/src/components/Form/LabeledErrorBoundInput.stories.tsx:23]: It looks like some spacing got removed here on line 23 and also on lines 28, 37, 45, 64, and 70, can you put them back for readability?

File: superset-frontend/src/components/Form/LabeledErrorBoundInput.tsx [hunk 4]
```diff
@@ -19,9 +19,9 @@
 import React from 'react';
 import { Input } from 'antd';
 import { styled, css, SupersetTheme } from '@superset-ui/core';
+import InfoTooltip from 'src/components/InfoTooltip';
 import FormItem from './FormItem';
 import FormLabel from './FormLabel';
-
```
Comment (lyndsiWilliams) [inline superset-frontend/src/components/Form/LabeledErrorBoundInput.tsx:24]: It looks like some spacing got removed here on line 24 and also on lines 37, 41, 49, 75 and 101, can you put them back for readability?

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx [hunk 5]
```diff
@@ -106,8 +108,23 @@ const CredentialsInfo = ({ changeMethods, isEditMode, db }: FieldPropTypes) => {
           </span>
         </div>
       ) : (
-        <div className="input-container">
-          <span className="label-select">Upload Credentials</span>
+        <div
+          className="input-container"
+          css={(theme: SupersetTheme) => infoTooltip(theme)}
+        >
+          {/*This is missing a reqired marker. Need clarification what element to add reqired to.*/}
```
Comment (lyndsiWilliams) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:115]: No worries here, I can fix it on my UI polish. Go ahead and remove this comment and I'll take care of the required bit 😁

**Review by lyndsiWilliams — CHANGES_REQUESTED** [review]
Just a couple of small nits, this looks great!

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx [hunk 6]
```diff
@@ -51,6 +51,8 @@ export const FormFieldOrder = [
 
 interface FieldPropTypes {
   required: boolean;
+  hasTooltip: boolean;
+  tooltipText: (valuse: any) => string;
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:55]: Are these required props?

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 7]
```diff
@@ -670,22 +672,30 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
...
+                          configuration_method:
+                            CONFIGURATION_METHOD.DYNAMIC_FORM,
+                          engine: db?.engine,
+                        },
+                      })
+                    }
+                    css={theme => alchemyButtonLinkStyles(theme)}
+                  >
+                    Connect this database using the dynamic form instead
+                    <InfoTooltip
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:692]: does putting this in the button component make the tooltip clickable? Was that intentional.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 8]
```diff
@@ -896,24 +906,30 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
...
+                          configuration_method:
+                            CONFIGURATION_METHOD.SQLALCHEMY_URI,
+                          database_name: db.database_name,
+                        },
+                      })
+                    }
+                    css={buttonLinkStyles}
+                  >
+                    Connect this database with a SQLAlchemy URI string instead
+                    <InfoTooltip
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:926]: Same here, you're surrounding the button + tooltip in a div, which makes me think you don't want the tooltip to be able to link to sql alchemy. But the info tooltip is still inside the button component.

**Review by andrewbastian — COMMENTED** [review]
- [x] Added spacing back into files where it was removed

- [x] Made `hasTooltip` and `tooltipText` optional props

- [x] Moved `<InfoTooltip/>`'s out of `LinkButtons`

- [x] Aligned tooltips with `viewBox`.

**Review by hughhhh — APPROVED** [review]

**Pull request merged**